readthedocs/api/v2/serializers.py

```diff
@@ -193,6 +194,19 @@ def get_matches(self, obj):
         if request.user is not None and request.user.is_authenticated:
             return obj.matches(request.user)
 
+    def is_admin(self, obj):
+        request = self.context['request']
+
+        # Use cached value
+        if hasattr(obj, 'admin'):
+            return obj.admin
```

Comment:
Can we use `functools.lru_cache` for this?

Reply:
I'm not that much familiar with ``functools.lru_cache`` but I tried it and it doesn't seem to work here. :( And I don't think we will have many subsequent calls to the same objects ``is_admin`` method in the API. But let me know what you think.

Reply:
Honestly, I don't really understand how the logic you write here works :sweat_smile:. I didn't see where we are setting `obj.admin` in our code to use it as a cached value. Can you point me to that?

Reply:
> I don't think we have any subsequent calls to the same objects `is_admin` method in the same API request

This is basically what I don't understand. If we don't have subsequent calls, why do we need to cache something here?

Reply:
`obj.admin` attribute is coming from the queryset annotation on the API view

https://github.com/readthedocs/readthedocs.org/blob/0c04f9f2db8763a64ecf9d600f4e3f1663b4d0e5/readthedocs/api/v2/views/model_views.py#L386

I went with this approach for database optimization. We require ``admin`` field in the API response and it is used in `import.js`

We have a few similar approaches in the codebase, for example, 

https://github.com/readthedocs/readthedocs.org/blob/7b53485714ba30ce5cfe63129eb9ab555f46abba/readthedocs/projects/models.py#L1176-L1188

Reply:
Thanks for the explanation! Your code is correct :smile: